flattentool/json_input.py

```diff
@@ -117,11 +119,33 @@ def __init__(self, json_filename=None, root_json_dict=None, schema_parser=None,
                 self.main_sheet.columns = []
                 for sheet_name, sheet in list(self.sub_sheets.items()):
                     sheet.columns = []
-            # Rollup is pulled from the schema_parser, as rollup is only possible if a schema parser is specified
-            self.rollup = schema_parser.rollup
             self.schema_parser = schema_parser
         else:
-            self.rollup = False
+            self.schema_parser = None
+
+        self.rollup = False
+        if rollup:
+            if schema_parser:
+                # If rollup is present in the schema this takes precedence over direct input.
+                self.rollup = schema_parser.rollup
+                if isinstance(rollup, (list,)) and (len(rollup) > 1 or (len(rollup) == 1 and rollup[0] is not True)):
+                    warn('Using rollUp values from schema, ignoring direct input.')
+            elif isinstance(rollup, (list,)):
+                if len(rollup) == 1 and os.path.isfile(rollup[0]):
```

Comment:
I agree that three separate options is a clearer CLI interface.

The command should error if more than one option is used, however.

Reply:
I deliberately kept only one multipurpose rollup option so the user wouldn't have to think about which argument to use for what is essentially the same functionality. But I can also see that separate options has its advantages. Does this change need to happen before this can be merged, or are we okay to bump it down the line? I can open a new issue for it if so.

Reply:
If we open a new issue, let's make sure we don't use 'backwards incompatibility' as a reason not to change it :)

Reply:
https://github.com/OpenDataServices/flatten-tool/issues/317